src/push.rs

```diff
@@ -215,6 +223,16 @@ pub async fn build_profile(data: PushProfileData<'_>) -> Result<(), PushProfileE
         &data.deploy_data.profile.profile_settings.path
     );
 
+    let nix_version_output = Command::new("nix").arg("--version")
+        .output()
+        .await.map_err(PushProfileError::NixVersion)?;
+
+    let version_regex = Regex::new(r"(\d+\.\d+\..+)").unwrap();
```

Comment:
I'm always a bet hesitant about regexes, this one assumes that the nix version output will always adhere to the `major.minor` format and fails if it doesn't :/

Reply:
Fair enough, do you have any ideas how can this be made more robust?

Reply:
Perhaps, we can try `^out` first and then fallback to the old format in case the first one fails

Reply:
Also, `nix path-info` might do the thing, but it'll require to parse output either stdout or stderr in case the resulting path doesn't exist yet

Reply:
Implemented an approach that uses `nix path-info`, which seems to be working and I believe it's a bit more robust than version grepping

Reply:
I would agree that `path-info` is probably much more robust